### PR TITLE
Add PayPalNativeFragment to Demo App 

### PR DESCRIPTION
**Pull request by mattwylder**
### Summary of changes

 - Added scaffolding for Native Checkout flow to Demo App

 ### Checklist

~- [ ] Added a changelog entry~

- @sarahkoop 
- @sshropshire 
- @mattwylder

### CODE REVIEW

File: Demo/src/main/java/com/braintreepayments/demo/MainFragment.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ public class MainFragment extends BaseFragment {
     private Button localPaymentsButton;
     private Button preferredPaymentMethodsButton;
     private Button samsungButton;
+    private Button payPalNativeButton;
```
Comment (jaxdesmarais) [inline Demo/src/main/java/com/braintreepayments/demo/MainFragment.java:34]:
```suggestion
    private Button payPalNativeCheckoutButton;
```

File: Demo/src/main/java/com/braintreepayments/demo/MainFragment.java [hunk 2]
```diff
@@ -45,6 +46,7 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         localPaymentsButton = view.findViewById(R.id.local_payment);
         preferredPaymentMethodsButton = view.findViewById(R.id.preferred_payment_methods);
         samsungButton = view.findViewById(R.id.samsung_pay);
+        payPalNativeButton = view.findViewById(R.id.paypal_native);
```
Comment (jaxdesmarais) [inline Demo/src/main/java/com/braintreepayments/demo/MainFragment.java:49]:
```suggestion
        payPalNativeCheckoutButton = view.findViewById(R.id.paypal_native_checkout);
```

File: Demo/src/main/java/com/braintreepayments/demo/MainFragment.java [hunk 3]
```diff
@@ -54,6 +56,7 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         venmoButton.setOnClickListener(this::launchVenmo);
         preferredPaymentMethodsButton.setOnClickListener(this::launchPreferredPaymentMethods);
         samsungButton.setOnClickListener(this::launchSamsungPay);
+        payPalNativeButton.setOnClickListener(this::launchPayPalNative);
```
Comment (jaxdesmarais) [inline Demo/src/main/java/com/braintreepayments/demo/MainFragment.java:59]:
```suggestion
        payPalNativeCheckoutButton.setOnClickListener(this::launchPayPalNativeCheckout);
```

File: Demo/src/main/res/layout/fragment_pay_pal_native.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:padding="10dp"
+    tools:context=".PayPalNativeFragment">
```
Comment (jaxdesmarais) [inline Demo/src/main/res/layout/fragment_pay_pal_native.xml:8]:
```suggestion
    tools:context=".PayPalNativeCheckoutFragment">
```

File: Demo/src/main/res/layout/fragment_pay_pal_native.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:orientation="vertical">
+
+        <Button
+            android:id="@+id/paypal_native_launch"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/paypal_native_launch" />
```
Comment (jaxdesmarais) [inline Demo/src/main/res/layout/fragment_pay_pal_native.xml:20]:
```suggestion
            android:text="@string/paypal_native_checkout_launch" />
```

File: Demo/src/main/res/navigation/nav_graph.xml [hunk 6]
```diff
@@ -34,6 +34,9 @@
         <action
             android:id="@+id/action_mainFragment_to_preferredPaymentMethodsFragment"
             app:destination="@id/preferredPaymentMethodsFragment" />
+        <action
+            android:id="@+id/action_mainFragment_to_payPalNativeFragment"
+            app:destination="@id/payPalNativeFragment" />
```
Comment (jaxdesmarais) [inline Demo/src/main/res/navigation/nav_graph.xml:39]:
```suggestion
            android:id="@+id/action_mainFragment_to_payPalNativeCheckoutFragment"
            app:destination="@id/payPalNativeCheckoutFragment" />
```

File: Demo/src/main/res/navigation/nav_graph.xml [hunk 7]
```diff
@@ -152,5 +155,10 @@
             app:popUpTo="@id/mainFragment"
             app:popUpToInclusive="false" />
     </fragment>
+    <fragment
+        android:id="@+id/payPalNativeFragment"
+        android:name="com.braintreepayments.demo.PayPalNativeFragment"
+        android:label="fragment_pay_pal_native"
+        tools:layout="@layout/fragment_pay_pal_native" />
```
Comment (jaxdesmarais) [inline Demo/src/main/res/navigation/nav_graph.xml:162]:
```suggestion
        android:id="@+id/payPalNativeCheckoutFragment"
        android:name="com.braintreepayments.demo.PayPalNativeCheckoutFragment"
        android:label="fragment_paypal_native_checkout"
        tools:layout="@layout/fragment_paypal_native_checkout" />
```

File: Demo/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -18,6 +18,7 @@
     <string name="local_payment_button">Local Payment</string>
     <string name="preferred_payment_methods_button">Pref Payment Methods</string>
     <string name="cards">Credit or Debit Cards</string>
+    <string name="paypal_native">PayPal Native</string>
```
Comment (jaxdesmarais) [inline Demo/src/main/res/values/strings.xml:21]:
```suggestion
    <string name="paypal_native_checkout">PayPal Native Checkout</string>
```

File: Demo/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -92,6 +93,7 @@
     <string name="paypal_display_name_summary">Name to be displayed in the PayPal flow</string>
     <string name="paypal_landing_page_type_billing">Billing</string>
     <string name="paypal_landing_page_type_login">Login</string>
+    <string name="paypal_native_launch">Launch PayPal Native</string>
```
Comment (jaxdesmarais) [inline Demo/src/main/res/values/strings.xml:96]:
```suggestion
    <string name="paypal_native_checkout_launch">Launch PayPal Native Checkout</string>
```

**Review by jaxdesmarais — COMMENTED** [review]
Since PayPal Native Checkout is the full product name, we should update the demo to ensure we are using the full name vs mixing PayPal Checkout and PayPal Native Checkout across the SDK and demo.

File: Demo/src/main/res/navigation/nav_graph.xml [hunk 10]
```diff
@@ -152,5 +155,10 @@
             app:popUpTo="@id/mainFragment"
             app:popUpToInclusive="false" />
     </fragment>
+    <fragment
+        android:id="@+id/payPalNativeCheckoutFragment"
+        android:name="com.braintreepayments.demo.PayPalNativeCheckoutFragment"
+        android:label="fragment_pay_pal_native_checkout"
+        tools:layout="@layout/fragment_pay_pal_native_checkout" />
```
Comment (jaxdesmarais) [inline Demo/src/main/res/navigation/nav_graph.xml:162]: `pay_pal` > `payPal` or `paypal`. Looks like we use both, not sure which is preferred on Android.

**Review by jaxdesmarais — APPROVED** [review]
Just one small cleanup comment, otherwise looks good

File: Demo/src/main/res/navigation/nav_graph.xml (hunk 10 above)
Comment (andrespelaezp) [reply Demo/src/main/res/navigation/nav_graph.xml:162]: I believe if we distinguish PayPal as two different words, then "pay_pal" would be okay, if not "paypal" would be better. I am not a big fan of camel case on xml

**Review by andrespelaezp — APPROVED** [review]

**Pull request merged**